### PR TITLE
dockerize apod-api

### PR DESCRIPTION
**Pull request by sanjeev-sreenath**
Enabling users to run the API as a Docker image locally

### CODE REVIEW

**Comment by JustinGOSSES** [comment]
Pease resubmit taking into consideration new code organization. Thank you for pull request and sorry for slow response. Would be nice to get a docker version in install instructions.

**Comment by sanjeev-sreenath** [comment]
> Pease resubmit taking into consideration new code organization. Thank you for pull request and sorry for slow response. Would be nice to get a docker version in install instructions.

Done.

**Comment by JustinGOSSES** [comment]
Did a quick look. @evantayloryates can you take a quick look at this, especially sure the change to application.py doesn't affect running locally and in aws. Thanks

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ lib/
 .DS_Store
 .idea/
 apod/__pycache__/
+venv/
```
Comment (AaronRobson) [inline .gitignore:13]: It is common practice to have a trailing newline at the end of files.

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
-FROM python:2-alpine
+FROM python:3-alpine
 
 WORKDIR /usr/src/app
 COPY requirements.txt ./
 RUN pip install --no-cache-dir -r requirements.txt
 COPY . .
-CMD [ "python", "apod/service.py" ]
+EXPOSE 5000
```
Comment (AaronRobson) [inline Dockerfile:7]: :+1: always good to be explicit about expected open ports.

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
+CMD python application.py
```
Comment (AaronRobson) [inline Dockerfile:8]: The list form is preferred according to: https://docs.docker.com/engine/reference/builder/#cmd

Please consider also this form:
```dockerfile
ENTRYPOINT ["python"]
CMD ["application.py"]
```
as discussed here: https://docs.docker.com/engine/reference/builder/#exec-form-entrypoint-example

File: application.py [hunk 4]
```diff
@@ -286,4 +286,7 @@ def application_error(e):
 
 
 if __name__ == '__main__':
-    app.run(host='0.0.0.0' ,port=5000)
+    application.run('0.0.0.0')
```
Comment (AaronRobson) [inline application.py:289]: I think that it would be best to be explicit about the port.

File: application.py [hunk 5]
```diff
@@ -286,4 +286,7 @@ def application_error(e):
...
+    # httpd = make_server('', 8000, application)
+    # print("Serving on port 8000...")
+    # httpd.serve_forever()
```
Comment (AaronRobson) [inline application.py:292]: I think you may have accidentally committed these commented out lines from a work in progress version.

File: README.md [hunk 6]
```diff
@@ -193,50 +213,50 @@ https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY&count=5
...
 [
   {
-    "copyright": "Panther Observatory", 
-    "date": "2006-04-15", 
-    "explanation": "In this stunning cosmic vista, galaxy M81 is on the left surrounded by blue spiral arms.  On the right marked by massive gas and dust clouds, is M82.  These two mammoth galaxies have been locked in gravitational combat for the past billion years.   The gravity from each galaxy dramatically affects the other during each hundred million-year pass.  Last go-round, M82's gravity likely raised density waves rippling around M81, resulting in the richness of M81's spiral arms.  But M81 left M82 with violent star forming regions and colliding gas clouds so energetic the galaxy glows in X-rays.  In a few billion years only one galaxy will remain.", 
-    "hdurl": "https://apod.nasa.gov/apod/image/0604/M81_M82_schedler_c80.jpg", 
-    "media_type": "image", 
-    "service_version": "v1", 
-    "title": "Galaxy Wars: M81 versus M82", 
+    "copyright": "Panther Observatory",
```
Comment (AaronRobson) [inline README.md:216]: Nice one for sorting out the trailing whitespace from these lines :+1:

**Pull request merged**